shared/.flowconfig

```diff
@@ -25,9 +39,23 @@ esproposal.class_static_fields=enable
 esproposal.class_instance_fields=enable
 experimental.const_params=true
 
-module.name_mapper='^image![a-zA-Z0-9$_-]+$' -> 'GlobalImageStub'
-module.name_mapper='^[./a-zA-Z0-9$_-]+\.\(bmp\|gif\|jpg\|jpeg\|png\|psd\|svg\|webp\|m4v\|mov\|mp4\|mpeg\|mpg\|webm\|aac\|aiff\|caf\|m4a\|mp3\|wav\|html\|pdf\)$' -> 'RelativeImageStub'
-module.name_mapper='^\..*resolve-root' -> 'resolveRoot'
+module.name_mapper='^[./a-zA-Z0-9$_-]+\.\(bmp\|gif\|jpg\|jpeg\|png\|psd\|svg\|webp\|m4v\|mov\|mp4\|mpeg\|mpg\|webm\|aac\|aiff\|caf\|m4a\|mp3\|wav\|html\|pdf\)$' -> '<PROJECT_ROOT>/__mocks__/global-image-stub.js'
+
+module.system=haste
+
+module.system.haste.use_name_reducers=true
+# strip .js or .js.flow suffix
+module.system.haste.name_reducers='^\(.*\)\.js\(\.flow\)?$' -> '\1'
+# strip .mobile suffix
+# module.system.haste.name_reducers='^\(.*\)\.ios$' -> '\1'
+# module.system.haste.name_reducers='^\(.*\)\.android$' -> '\1'
+# module.system.haste.name_reducers='^\(.*\)\.native$' -> '\1'
+# module.system.haste.name_reducers='^\(.*\)\.desktop\.js$' -> '\1'
```

Comment:
should this include the `.js`?

Reply:
i think the other mapper runs first actually. anyways its commented out for a later phase. we'll figure it out then (this came from RN's .flowconfig actually)